codec-http/src/main/java/io/netty/handler/codec/http/cookie/ServerCookieDecoder.java

```diff
@@ -157,20 +127,53 @@
                     if (i == headerLen) {
                         // NAME (no value till the end of string)
                         newNameEnd = headerLen;
-                        value = null;
+                        newValueBegin = newValueEnd = -1;
                         break;
                     }
                 }
             }
 
-            if (!rfc2965Style || (!header.regionMatches(newNameStart, "$Path", 0, "$Path".length()) &&
-                    !header.regionMatches(newNameStart, "$Domain", 0, "$Domain".length()) &&
-                    !header.regionMatches(newNameStart, "$Port", 0, "$Port".length()))) {
+            if (newValueBegin == -1) {
+                logger.debug("Skipping cookie with null value");
+                continue;
+            }
+
+            if (rfc2965Style && (header.regionMatches(newNameBegin, RFC2965_PATH, 0, RFC2965_PATH.length()) ||
+                    header.regionMatches(newNameBegin, RFC2965_DOMAIN, 0, RFC2965_DOMAIN.length()) ||
+                    header.regionMatches(newNameBegin, RFC2965_PORT, 0, RFC2965_PORT.length()))) {
 
                 // skip obsolete RFC2965 fields
-                String name = header.substring(newNameStart, newNameEnd);
-                cookies.add(new DefaultCookie(name, value));
+                continue;
+            }
+
+            final String name = header.substring(newNameBegin, newNameEnd);
+
+            int invalidOctetPos;
+            if ((invalidOctetPos = firstInvalidCookieNameOctet(name)) >= 0) {
+                logger.debug("Skipping cookie because name '{}' contains invalid char '{}'",
+                        name, name.charAt(invalidOctetPos));
+                continue;
+            }
+
+            CharSequence wrappedValue = CharBuffer.wrap(header, newValueBegin, newValueEnd);
+            CharSequence unwrappedValue = unwrapValue(wrappedValue);
+            if (unwrappedValue == null) {
+                logger.debug("Skipping cookie because starting quotes are not properly balanced in '{}'",
+                        wrappedValue);
+                continue;
             }
+
+            final boolean wrap = unwrappedValue.length() != newValueEnd - newValueEnd;
+
+            if ((invalidOctetPos = firstInvalidCookieValueOctet(unwrappedValue)) >= 0) {
+                logger.debug("Skipping cookie because value '{}' contains invalid char '{}'",
+                        unwrappedValue, unwrappedValue.charAt(invalidOctetPos));
+                continue;
```

Comment:
@normanmaurer Should I make the new Encoders/Decoders objects instead of a bunch of static methods, with some boolean option passed in the constructor? Or do you have a standard option mechanism, like System props, that you want me to use?